django/db/backends/postgresql/schema.py

```diff
@@ -255,6 +262,28 @@ def _alter_column_type_sql(
                 model, old_field, new_field, new_type, old_collation, new_collation
             )
 
+    def _new_index_should_be_added(self, old_field, new_field):
+        return not (old_field.db_index or old_field.unique) and (
+            new_field.db_index or new_field.unique
+        )
+
+    def _deleted_index_should_be_recreated(
+        self, old_field, new_field, old_type, new_type
+    ):
+        if (
+            old_field.db_index
+            and not old_field.unique
+            and (
+                not new_field.db_index
+                or (new_field.unique and not new_field.primary_key)
+            )
```

Comment:
Thanks, I added a for a `db_index` being removed from a field, and removed the `old_field.db_index` check. I think it may have come from [https://github.com/django/django/blob/main/django/db/backends/base/schema.py#L1011](https://github.com/django/django/blob/main/django/db/backends/base/schema.py#L1011), but may not be needed here.